### PR TITLE
[IIIF-437] Update queuing mechanism; add documentation

### PR DESCRIPTION
**Pull request by ksclarke**
* Update queuing mechanism from reliance on client pool to an explicit config/counter for maximum number of S3 requests
* Add documentation
* Refactor MainVerticle to remove one layer of nesting in router setup

The default values for the queuing configs will be changed once we've done some experimentation with adjusting them to the system on which we'll be running.

### CODE REVIEW

File: src/main/java/edu/ucla/library/bucketeer/verticles/MainVerticle.java [hunk 1]
```diff
@@ -47,73 +50,91 @@
...
+     * Gets the available logical processors on the system minus one.
+     *
+     * @return
+     */
+    private static int getAvailableProcessors() {
+        final SystemInfo systemInfo = new SystemInfo();
+        final HardwareAbstractionLayer hardwareAbstractionLayer = systemInfo.getHardware();
+        final CentralProcessor centralProcessor = hardwareAbstractionLayer.getProcessor();
+
+        return centralProcessor.getLogicalProcessorCount() - 1;
```
Comment (cachemeoutside) [inline src/main/java/edu/ucla/library/bucketeer/verticles/MainVerticle.java]: Seems like we're going to -1 core route where we'd want to leave a core free for processing/adding items into queues?

Comment (cachemeoutside) [reply src/main/java/edu/ucla/library/bucketeer/verticles/MainVerticle.java]: wondering if there's an edge case here where we might get a 0 - 1 for core count.

Comment (ksclarke) [reply src/main/java/edu/ucla/library/bucketeer/verticles/MainVerticle.java]: As a default, I went core -1. I think we need to try more threads as an option though. I'm not sure whether the cost of thread management will outweigh the additional benefit of thread swapping. I can't imagine a machine that has zero logical cores. I could imagine a single core machine, but not sure what the chance of that would be.

File: README.md [hunk 2]
```diff
@@ -41,6 +50,25 @@ You can run a development instance of Bucketeer by typing the following within t
...
+  <dt>The maximum number of S3 requests</dt>
+  <dd>If you set the above two values very high, it's easy to run out of RAM on your machine (since the S3 clients read the TIFF files concurrently). The maximum number of S3 requests threshold provides an upper limit on the number of S3 PUTs that can be in process at any given time, regardless of the number of worker verticles or threads that have been configured.<br/><br/>The maximum number of requests you allow will depend on the amount of RAM available on the machine. When the maximum has been reached, conversion requests are requeued until there are resources available. The property for this configuration is <code>s3.max.requests</code>.<br/><br/>The ways that it can be set are just like the prior two properties.</dd>
```
[the remark this answers is not in the document]
Comment (ksclarke) [reply README.md]: Yes, that's correct. There is no attempt to catch OOMs and adjust from there. It's up to the user at this point to do some trial and error and select a value that's going to work for their particular amount of resources. Might be nice to try to do some automatic adjustments in the future, but for now the responsibility is on the user.

**Pull request merged**